### PR TITLE
[TASK] refactor header/footer

### PR DESCRIPTION
**Pull request by JimHolman1199**
Refactor header/footer
Write tests to header/footer
Write tests to search-item/search-not-found components

### CODE REVIEW

File: src/app/component/layout/components/footer/footer.component.scss [hunk 1]
```diff
@@ -1,199 +1,121 @@
...
+    width: fit-content;
+    margin: 0 auto;
   }
 }
 
-@media screen and (max-width: 576px) {
-  .main {
-    margin: 0 2.4%;
+.footer_bottom-part {
+  color: #878787;
```
Comment (luchward) [inline src/app/component/layout/components/footer/footer.component.scss:64]: It should be a variable

File: src/app/component/layout/components/header/header.component.ts [hunk 2]
```diff
@@ -59,9 +57,7 @@ export class HeaderComponent implements OnInit {
   ngOnInit() {
     this.searchSearch.searchSubject.subscribe(signal => this.openSearchSubscription(signal));
     this.searchSearch.allSearchSubject.subscribe(signal => this.openAllSearchSubscription(signal));
```
Comment (luchward) [inline src/app/component/layout/components/header/header.component.ts:59]: Why did you leave subscriptions without unsubscription?

**Review by luchward — APPROVED** [review]

File: src/app/component/layout/components/footer/footer.component.html [hunk 3]
```diff
@@ -1,78 +1,60 @@
...
-              </a>
-            </li>
-            <li>
-              <a [routerLink]="['/profile', getUserId()]"
-                 href=""
-                 class="router-links">{{ "user.footer.my-habits" | translate }}
-              </a>
-            </li>
-          </ul>
+  <div class="footer_background-image">
```
Comment (skrebetsnataliia) [inline src/app/component/layout/components/footer/footer.component.html:2]: Why did you use BEM methodology? According to project's style guide we should use dashes in class names, isn't it?

**Review by skrebetsnataliia — APPROVED** [review]

File: src/app/component/layout/components/footer/footer.component.html (hunk 3 above)
Comment (JimHolman1199) [reply src/app/component/layout/components/footer/footer.component.html:2]: it's the same for me.
just BEM expands it..

File: src/app/component/layout/components/footer/footer.component.ts [hunk 4]
```diff
@@ -1,26 +1,31 @@
...
+    this.userIdSub = this.localStorageService
+      .userIdBehaviourSubject.subscribe(userId => this.userId = userId);
   }
 
   public getUserId(): number | string {
     return ((this.userId !== null && !isNaN(this.userId)) ? this.userId : 'not_signed-in');
   }
 
+  ngOnDestroy() {
+    this.userIdSub.unsubscribe();
```
Comment (mr-pavlenko) [inline src/app/component/layout/components/footer/footer.component.ts:29]: use take untill

**Review by mr-pavlenko — APPROVED** [review]

**Pull request merged**